purepyvmware/datastores.py

```diff
@@ -0,0 +1,329 @@
+import time
+
+from lib import datastore_utils
+from pyVmomi import vim
+from pyVmomi import sms
+
+
+class Datastores(object):
+    """Maintenance of Pure Storage based datastores and associated array volumes.
+
+    For general maintenance of VMware datastores that are backed by Pure Storage FlashArrays.
+    """
+
+    def __init__(self, vsphere_content, flasharray):
+        """Setting up vSphere and FlashArray environment for use.
+
+        Args:
+            vsphere_content (vim.ServiceInstanceContent): All available content within the vSphere env.
+            flasharray (purestorage.FlashArray): Existing session to FlashArray.
+        """
+        self.flasharray = flasharray
+        self.pure_identifier = '24a937'
+        self.vsphere_content = vsphere_content
+
+    def create_compute_container_view(self):
+        """Create compute container view for an inventory of managed object references.
+
+        Returns:
+            resources (vim.view.ContainerView): Object view of compute resources.
+        """
+        compute_container = self.vsphere_content.viewManager.CreateContainerView(self.vsphere_content.rootFolder,
+                                                                                 [vim.ComputeResource], True)
+        compute_resources = compute_container.view
+        compute_container.Destroy()
+
+        return compute_resources
+
+    def create_datastore_container_view(self):
+        """Create datastore container view for an inventory of managed object references.
+
+        Returns:
+            datastores (vim.view.ContainerView): Object view of datastore resources.
+        """
+        datastore_container = self.vsphere_content.viewManager.CreateContainerView(self.vsphere_content.rootFolder,
+                                                                                   [vim.Datastore], True)
+        datastores = datastore_container.view
+        datastore_container.Destroy()
```

Comment:
I don't know the library you're using, but won't any object not returned by the function be destroyed by garbage collection?